### PR TITLE
Composable mint schedules functions

### PR DESCRIPTION
**Pull request by PabloSzx**
feel free to propose better naming

### CODE REVIEW

**Review by saihaj — APPROVED** [review]

File: packages/sdk/src/client.ts [hunk 1]
```diff
@@ -541,23 +558,58 @@ export function SoundClient({
     return Array.from(allMinters)
   }
 
-  // Minting information from a minting contract for a given edition
-  async function _mintInfosFromMinter({
+  async function contractMinterMintIds({
```
Comment (gigamesh) [inline packages/sdk/src/client.ts:561]:
```suggestion
  async function editionMintIds({
```

File: packages/sdk/src/client.ts [hunk 2]
```diff
@@ -541,23 +558,58 @@ export function SoundClient({
...
 
     // Query MintConfigCreated event
     const minterContract = IMinterModule__factory.connect(minterAddress, signerOrProvider)
     const filter = minterContract.filters.MintConfigCreated(editionAddress)
     const mintScheduleConfigEvents = await minterContract.queryFilter(filter, fromBlockOrBlockHash)
-    const mintIds = mintScheduleConfigEvents.map((event) => event.args.mintId)
+    return mintScheduleConfigEvents.map((event) => event.args.mintId.toNumber())
+  }
+
+  async function contractMintSchedules({
```
Comment (gigamesh) [inline packages/sdk/src/client.ts:579]:
```suggestion
  async function mintSchedulesById({
```

File: packages/sdk/src/client.ts [hunk 3]
```diff
@@ -496,12 +517,8 @@ export function SoundClient({
     return networkChain === chainId
   }
 
-  /*********************************************************
-                  INTERNAL FUNCTIONS
- ********************************************************/
-
   // Addresses with MINTER_ROLE for a given edition
-  async function _registeredMinters({
+  async function contractRegisteredMinters({
```
Comment (gigamesh) [inline packages/sdk/src/client.ts:521]:
```suggestion
  async function registeredMinters({
```

**Review by gigamesh — COMMENTED** [review]
If we need a prefix for function name, I'd go with `edition` over `contract`. But we can probably omit altogether when the edition address is a required argument.

File: packages/sdk/src/client.ts (hunk 3 above)
Comment (PabloSzx) [reply packages/sdk/src/client.ts]: https://github.com/soundxyz/sdk/pull/126/commits/b137afd7b165e3dc8edfb0560a6b68419878aa85

File: packages/sdk/src/client.ts (hunk 2 above)
Comment (PabloSzx) [reply packages/sdk/src/client.ts]: https://github.com/soundxyz/sdk/pull/126/commits/b137afd7b165e3dc8edfb0560a6b68419878aa85

File: packages/sdk/src/client.ts (hunk 1 above)
Comment (PabloSzx) [reply packages/sdk/src/client.ts]: https://github.com/soundxyz/sdk/pull/126/commits/b137afd7b165e3dc8edfb0560a6b68419878aa85

File: .changeset/many-suns-leave.md [hunk 4]
```diff
@@ -0,0 +1,5 @@
+---
+'@soundxyz/sdk': minor
+---
+
+Composable mint schedules functions with "contractRegisteredMinters","contractMinterMintIds" and "contractMintSchedules"
```
Comment (PabloSzx) [inline .changeset/many-suns-leave.md:5]:
```suggestion
Composable mint schedules functions with "editionRegisteredMinters","editionMinterMintIds", "editionScheduleIds" and "editionMintSchedules"
```

**Review by saihaj — APPROVED** [review]

**Review by vigneshka — APPROVED** [review]

**Pull request merged**